test/unit/org/apache/cassandra/dht/BootStrapperTest.java

```diff
@@ -83,6 +104,44 @@ public void testSourceTargetComputation() throws UnknownHostException
         }
     }
 
+    @Test
+    @BMRule(name = "Make sure the non-optimized path is picked up for some operations",
+    targetClass = "org.apache.cassandra.dht.RangeStreamer",
+    targetMethod = "convertPreferredEndpointsToWorkMap(EndpointsByReplica)",
+    action = "org.apache.cassandra.dht.BootStrapperTest.optimizationSkipped()")
+    public void testOptmizationsSkip() throws UnknownHostException
+    {
+        testSkipFFOptmizationFeatureFlag(true);
```

Comment:
:comment
I think it would be worth to test that not setting a flag does not break the logic for cases where the optimized streaming candidates have to be calculated.

Reply:
When the flag is not set or false there can be cases the optimization is hit and others where it isn't. I can add a check on that case that both optimized and non-optimized are hit.